integrations/optimum/src/haystack_integrations/components/embedders/optimum_text_embedder.py

```diff
@@ -0,0 +1,200 @@
+            self,
+            model=self.model,
+            prefix=self.prefix,
+            suffix=self.suffix,
+            normalize_embeddings=self.normalize_embeddings,
+            onnx_execution_provider=self.onnx_execution_provider,
+            model_kwargs=self.model_kwargs,
+            token=self.token.to_dict() if self.token else None,
+        )
+
+        model_kwargs = serialization_dict["init_parameters"]["model_kwargs"]
+        model_kwargs.pop("token", None)
+
+        serialize_hf_model_kwargs(model_kwargs)
+        return serialization_dict
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "OptimumTextEmbedder":
+        """
+        Deserialize this component from a dictionary.
+        """
+        deserialize_secrets_inplace(data["init_parameters"], keys=["token"])
+        deserialize_hf_model_kwargs(data["init_parameters"]["model_kwargs"])
+        return default_from_dict(cls, data)
+
+    def mean_pooling(self, model_output: torch.tensor, attention_mask: torch.tensor) -> torch.tensor:
+        """
+        Perform Mean Pooling on the output of the Embedding model.
+
+        :param model_output: The output of the embedding model.
+        :param attention_mask: The attention mask of the tokenized text.
+        :return: The embeddings of the text after mean pooling.
+        """
+        # First element of model_output contains all token embeddings
+        token_embeddings = model_output[0]
+        input_mask_expanded = attention_mask.unsqueeze(-1).expand(token_embeddings.size()).float()
+        sum_embeddings = torch.sum(token_embeddings * input_mask_expanded, 1)
+        sum_mask = torch.clamp(input_mask_expanded.sum(1), min=1e-9)
+        return sum_embeddings / sum_mask
+
+    @component.output_types(embedding=List[float])
+    def run(self, text: str):
+        """Embed a string.
+
+        :param text: The text to embed.
+        :return: The embeddings of the text.
+        """
+        if not isinstance(text, str):
+            msg = (
+                "OptimumTextEmbedder expects a string as an input. "
+                "In case you want to embed a list of Documents, please use the OptimumDocumentEmbedder."
+            )
+            raise TypeError(msg)
+
+        if not (self.embedding_model and self.tokenizer):
+            msg = "The embedding model has not been loaded. Please call warm_up() before running."
+            raise RuntimeError(msg)
+
+        text_to_embed = self.prefix + text + self.suffix
+
+        # Determine device for tokenizer output
+        device = (
+            "cuda"
+            if self.onnx_execution_provider
+            in ["CUDAExecutionProvider", "ROCMExecutionProvider", "TensorrtExecutionProvider"]
+            else "cpu"
+        )
+
+        encoded_input = self.tokenizer([text_to_embed], padding=True, truncation=True, return_tensors="pt").to(device)
+
+        # Compute token embeddings
+        with torch.no_grad():
```

Comment:
I believe it is generally recommend to use `torch.inference_mode()` instead of `torch.no_grad()` since I think it contains additional speed ups. 

However, I could be wrong here and that this does not mesh well with ONNX runtime. Do you know if we should switch to `torch.inference_mode()`?

Reply:
Now that you mention it, I think it's a no-op in either case as PyTorch won't be used to execute the model. The tokenizer's encode operation could still make use of it, but I'd wager that the cost/memory savings are negligible at best. 

We can just remove it.

Reply:
You're right, I will remove it, its not required here.